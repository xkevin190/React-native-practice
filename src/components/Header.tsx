import * as React from 'react';
import {
  Text,
  View,
  StyleSheet,
  StatusBar,
  TouchableOpacity,
} from 'react-native';
import {NavigationProps} from '../constants/types';
import {normalize, SCREEN_HEIGHT, SCREEN_WIDTH} from '../constants/utils';
import Icon from 'react-native-vector-icons/Ionicons';
import {colors} from '../constants/colors';

type headerProps = {
  navigation?: NavigationProps;
  title: string;
};


const Header = (props: headerProps) => {
  const goToback = () => {
    props.navigation.openDrawer();
  };

  return (
    <View style={styles.container}>
      <StatusBar backgroundColor={colors.background} />
      <TouchableOpacity onPress={goToback}>
        <Icon color={colors.primary} name="menu" size={normalize(23)} />
      </TouchableOpacity>
      <View>
        <Text style={styles.TextHeader}>{props.title}</Text>
      </View>
      <View style={{paddingRight: SCREEN_WIDTH * 0.04}} />
    </View>
  );
};

export default Header;

const styles = StyleSheet.create({
  container: {
    backgroundColor: colors.background,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingHorizontal: SCREEN_WIDTH * 0.04,
    height: SCREEN_HEIGHT * 0.06,
  },
  TextHeader: {
    fontFamily: 'Urbanist',
    fontSize: normalize(20),
    fontWeight: '600',
    color: colors.primary,
  },
});
