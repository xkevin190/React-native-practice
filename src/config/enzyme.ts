import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

/**
 * @tech-debt
 *
 * Update enzyme to support react 18
 */
Enzyme.configure({ adapter: new Adapter() });
