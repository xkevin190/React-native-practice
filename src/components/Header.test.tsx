import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';
import Header from './Header';

describe('Header (Unit-Test)', () => {
  let wrapper: ShallowWrapper;

  const navigation = {
    navigate: jest.fn(),
    openDrawer: jest.fn(),
  };

  beforeEach(() => {
    wrapper = shallow(<Header navigation={navigation} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('openDrawer function should executed', () => {
    wrapper.find('TouchableOpacity').first().simulate('press');
    expect(navigation.openDrawer).toBeCalled();
  });
});
