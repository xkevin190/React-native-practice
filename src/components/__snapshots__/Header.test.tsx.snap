// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header (Unit-Test) should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#17141f",
      "flexDirection": "row",
      "height": 80.03999999999999,
      "justifyContent": "space-between",
      "paddingHorizontal": 30,
    }
  }
>
  <StatusBar
    backgroundColor="#17141f"
  />
  <TouchableOpacity
    onPress={[Function]}
  >
    <Icon
      allowFontScaling={false}
      color="#ffffff"
      name="menu"
      size={52}
    />
  </TouchableOpacity>
  <View>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontFamily": "Urbanist",
          "fontSize": 45,
          "fontWeight": "600",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "paddingRight": 30,
      }
    }
  />
</View>
`;
